pkg/proxyagent/agent/agent.go

```diff
@@ -337,7 +337,7 @@ func managedProxyServiceResolverToFilterServiceToExpose(serviceResolvers []proxy
 func convertManagedProxyServiceResolverToService(clusterName string, sr proxyv1alpha1.ManagedProxyServiceResolver) serviceToExpose {
 	return serviceToExpose{
 		Host:         util.GenerateServiceURL(clusterName, sr.Spec.ServiceSelector.ServiceRef.Namespace, sr.Spec.ServiceSelector.ServiceRef.Name),
-		ExternalName: fmt.Sprintf("%s.%s.svc", sr.Spec.ServiceSelector.ServiceRef.Name, sr.Spec.ServiceSelector.ServiceRef.Namespace),
+		ExternalName: fmt.Sprintf("%s.%s.svc.cluster.local", sr.Spec.ServiceSelector.ServiceRef.Name, sr.Spec.ServiceSelector.ServiceRef.Namespace),
```

Comment:
why cluster.local?

Reply:
From the test, it seems that this a necessary suffix. But I need to make a deep invest to answer why... I will bring the result to this thread later.

Reply:
Found in coreDNS document, says it's not a recursive resolver: https://coredns.io/manual/setups/#recursive-resolver

Reply:
But in go dns client, it assumes that dns servers in resolv.conf should be a recursive-resolver.  https://github.com/golang/go/blob/6f445a9db55f65e55c5be29d3c506ecf3be37915/src/net/dnsclient_unix.go#L666